WordPress/Classes/Services/SharingService.swift

```diff
@@ -0,0 +1,413 @@
+            services = []
+        }
+
+        return services
+    }
+
+
+    // MARK: - Private PublicizeService Methods
+
+
+    /**
+    Called when syncing Publicize services. Merges synced and cached data, removing
+    anything that does not exist on the server.  Saves the context.
+
+    @param remoteServices An array of `RemotePublicizeService` objects to merge.
+    @param success An optional callback block to be performed when core data has saved the changes.
+    */
+    private func mergePublicizeServices(remoteServices:[RemotePublicizeService], success:(() -> Void)? ) {
+        managedObjectContext.performBlock { () -> Void in
+            let currentPublicizeServices = self.allPublicizeServices()
+
+            // Create or update based on the contents synced.
+            let servicesToKeep = remoteServices.map { (let remoteService) -> PublicizeService in
+                let pubService = self.createOrReplaceFromRemotePublicizeService(remoteService)
+                return pubService
+            }
+
+            // Delete any cached PublicizeServices that were not synced.
+            if currentPublicizeServices.count > 0 {
+                for pubService in currentPublicizeServices {
+                    if !servicesToKeep.contains(pubService) {
+                        self.managedObjectContext.deleteObject(pubService)
+                    }
+                }
+            }
+
+            // Save all the things.
+            ContextManager.sharedInstance().saveContext(self.managedObjectContext, withCompletionBlock: { () -> Void in
+                success?()
+            })
+        }
+    }
+
+
+    /**
+    Composes a new `PublicizeService`, or updates an existing one, with data represented by the passed `RemotePublicizeService`.
+
+    @return A `PublicizeService`.
+    */
+    private func createOrReplaceFromRemotePublicizeService(remoteService:RemotePublicizeService) -> PublicizeService {
+        var pubService = findPublicizeServiceNamed(remoteService.serviceID)
+        if pubService == nil {
+            pubService = NSEntityDescription.insertNewObjectForEntityForName(PublicizeService.classNameWithoutNamespaces(),
+                inManagedObjectContext: managedObjectContext) as? PublicizeService
+        }
+        pubService?.connectURL = remoteService.connectURL
+        pubService?.detail = remoteService.detail
+        pubService?.icon = remoteService.icon
+        pubService?.jetpackModuleRequired = remoteService.jetpackModuleRequired
+        pubService?.jetpackSupport = remoteService.jetpackSupport
+        pubService?.label = remoteService.label
+        pubService?.multipleExternalUserIDSupport = remoteService.multipleExternalUserIDSupport
+        pubService?.order = remoteService.order
+        pubService?.serviceID = remoteService.serviceID
+        pubService?.type = remoteService.type
+
+        return pubService!
```

Comment:
Just wondering if we should return an optional, instead of forcing the unwrap. It's probably okay to keep the `!`